packages/babel-cli/src/babel/util.ts

```diff
@@ -15,22 +15,54 @@ export function chmod(src: string, dest: string): void {
   }
 }
 
+export function alphasort(a: string, b: string) {
+  return a.localeCompare(b, "en");
+}
+
 type ReaddirFilter = (filename: string) => boolean;
 
 export function readdir(
   dirname: string,
   includeDotfiles: boolean,
   filter?: ReaddirFilter,
 ): Array<string> {
-  return readdirRecursive(dirname, (filename, index, currentDirectory) => {
-    const stat = fs.statSync(path.join(currentDirectory, filename));
-
-    if (stat.isDirectory()) return true;
-
+  if (process.env.BABEL_8_BREAKING) {
     return (
-      (includeDotfiles || filename[0] !== ".") && (!filter || filter(filename))
+      fs
+        .readdirSync(dirname, { recursive: true, withFileTypes: true })
+        .filter(dirent => {
+          // exclude directory entries from readdir results
+          if (dirent.isDirectory()) return false;
+          const filename = dirent.name;
+          return (
+            (includeDotfiles || filename[0] !== ".") &&
+            (!filter || filter(filename))
+          );
+        })
+        .map(dirent => path.join(dirent.path, dirent.name))
```

Comment:
The `Dirent#path` has been deprecated in favour of `Dirent#parentPath`, however to use `parentPath` we will have to further bump Node.js requirements to `^18.20.0 || ^20.12.0 || >= 21.4.0`. I would like to hear about your opinions.

Reply:
This all seems ok to me.

Reply:
OK, I will open a new PR to bump node and replace `Dirent#path` usage once the node typings are ready: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69490